### ISSUE TITLE
Automatically accept/reject PR after 15 minutes and 75% approval

### ISSUE THREAD

**Issue by Prior99** [botwillacceptanything/botwillacceptanything#347]
If a PR was open for 15 minutes and has >75% or < 25%  approval rate it can automatically be closed:

 * If 6 people vote :+1: (>75%), even if two more people vote :-1: the PR will succeed
 * if 6 people vote :-1: (<25%), if if two more people vote :+1: the PR will fail

So after 6 people voted the same, the outcome is clear and the PR can be accepted or rejected earlier.

**Comment by thenightwassave** [comment]
I was discussing this with the IRC room yesterday and here is what I concluded to:

REQUIRED_SUPERMAJORITY does not make sense in a context of an unbounded number of voters.

Right now we wait to decide a vote only when PERIOD has passed and MIN_VOTES has been reached. This results in things like 6/8 votes one way (when the REQUIRED_SUPERMAJORITY is at 65%) not passing right away even though there is _no other way_ for it to end.

I propose the correct method of handling this is to end the vote only when PERIOD has ended, and a REQUIRED_SUPERMAJORITY of MIN_VOTES is achieved either way. This would mean a 6/8 vote would result in a decision once PERIOD passes and eliminating the above problem.

Here is a quick diff showing how it would work: https://github.com/thenightwassave/botwillacceptanything/commit/a41054c3bac5112050c85395060068edeec9938a#diff-0808401482a44ccb945b82f8b6148385

Technically with this scheme REQUIRED_MAJORTY and MIN_VOTES just compute one value that we should use instead: MINIMUM_WINNING_VOTES. Right now with out defaults that would be 6. If yea or nea is at MINIMUM_WINNING_VOTES and PERIOD has passed, then the vote can end. Otherwise, vote stays open.

**Comment by Prior99** [comment]
So, vote stays open until either 6 people vote yes or 6 people vote no?

**Comment by thenightwassave** [comment]
That is correct, but I just defined it in terms of the variables we currently use.

**Comment by thenightwassave** [comment]
Would have to reword the bot vote starting comment too to explain the process.

**Comment by SpenserJ** [comment]
However, it could reach 6/6 in that timespan, at which point it would stay open for voting.

**Comment by SpenserJ** [comment]
@Prior99 Does the change a few days ago (close when there is a guaranteed result) get closer to what you wanted in your initial post?
